builtin/credential/aws-ec2/path_login.go

```diff
@@ -126,7 +127,7 @@ func validateMetadata(clientNonce, pendingTime string, storedIdentity *whitelist
 	//
 	// This is a weak criterion and hence the `allow_instance_migration` option
 	// should be used with caution.
-	if clientNonce != storedIdentity.ClientNonce {
+	if subtle.ConstantTimeCompare([]byte(clientNonce), []byte(storedIdentity.ClientNonce)) == 0 {
```

Comment:
The docs only say that the result will be 1 if the two are identical, not what the result will be if they're not. You should do `!= 1` instead. I realize that the code seems like it will always return 0, but the Go authors are free to change it so long as the behavior matches the docs.